core/commands/diff.py

```diff
@@ -571,54 +575,178 @@ class GsDiffOpenFileAtHunkCommand(TextCommand, GitCommand):
         """
         Show file at target commit if `git_savvy.diff_view.target_commit` is non-empty.
         Otherwise, open the file directly.
         """
         target_commit = self.view.settings().get("git_savvy.diff_view.target_commit")
         full_path = os.path.join(self.repo_path, filename)
+        window = self.view.window()
+        if not window:
+            return
+
         if target_commit:
-            self.view.window().run_command("gs_show_file_at_commit", {
+            window.run_command("gs_show_file_at_commit", {
                 "commit_hash": target_commit,
                 "filepath": full_path,
-                "lineno": lineno
+                "lineno": row,
             })
         else:
-            self.view.window().open_file(
-                "{file}:{row}:{col}".format(file=full_path, row=lineno, col=0),
+            window.open_file(
+                "{file}:{row}:{col}".format(file=full_path, row=row, col=col),
                 sublime.ENCODED_POSITION
             )
 
+    def jump_position_to_file(self, diff, pt):
+        # type: (ParsedDiff, int) -> Optional[Tuple[str, int, int]]
+        head_and_hunk_offsets = head_and_hunk_for_pt(diff, pt)
+        if not head_and_hunk_offsets:
+            return None
+
+        view = self.view
+        header_region, hunk_region = head_and_hunk_offsets
+        header = extract_content(view, header_region)
+        hunk = extract_content(view, hunk_region)
+        hunk_start, _ = hunk_region
+
+        rowcol = real_rowcol_in_hunk(hunk, relative_rowcol_in_hunk(view, hunk_start, pt))
+        if not rowcol:
+            return None
+
+        row, col = rowcol
+
+        filename = extract_filename_from_header(header)
+        if not filename:
+            return None
+
+        return filename, row, col
+
+
+def relative_rowcol_in_hunk(view, hunk_start, pt):
+    # type: (sublime.View, Point, Point) -> RowCol
+    """Return rowcol of given pt relative to hunk start"""
+    head_row, _ = view.rowcol(hunk_start)
+    pt_row, col = view.rowcol(pt)
+    # If `col=0` the user is on the meta char (e.g. '+- ') which is not
+    # present in the source. We pin `col` to 1 because the target API
+    # `open_file` expects 1-based row, col offsets.
+    return pt_row - head_row, max(col, 1)
+
+
+def real_rowcol_in_hunk(hunk, relative_rowcol):
+    # type: (str, RowCol) -> Optional[RowCol]
+    """Translate relative to absolute row, col pair"""
+    hunk_lines = split_hunk(hunk)
+    if not hunk_lines:
+        return None
+
+    row_in_hunk, col = relative_rowcol
+
+    # If the user is on the header line ('@@ ..') pretend to be on the
+    # first visible line with some content instead.
+    if row_in_hunk == 0:
+        row_in_hunk = next(
+            (
+                index
+                for index, line in enumerate(hunk_lines, 1)
+                if line.mode in ('+', ' ') and line.text.strip()
+            ),
+            1
+        )
```

Comment:
As I understand these lines of code. Get me the index of the first line with  + or - and the like text in not blank. Why the not blank check?

Reply:
This is basically an arbitrary choice. As I tested this it just felt better if the cursor jumped to a position with code on it. Iirc I added this as a separate commit so it was literally fine tuning the algorithm.

These edge case decisions are why I say mostly accurate jumping. Most of the time it is accurate jumping t.i. fixing the previous buggy behavior. And then I added two arbitrary decisions which I think feel a tiny bit better.